.circleci/config.yml

```diff
@@ -0,0 +1,161 @@
+
+      - run:
+          name: Upload a Python package to Pypi
+          command: |
+            source /tmp/venv/openfisca_france/bin/activate
+            .circleci/publish-python-package.sh
+
+      - run:
+          name: Publish a git tag
+          command: .circleci/publish-git-tag.sh
+
+      - run:
+          name: Deploy the Web API
+          command: ssh -o StrictHostKeyChecking=no deploy-api@fr.openfisca.org
+
+  build_python3:
+    parallelism: 3
+    docker:
+      - image: python:3.6
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+
+      - run:
+          name: Create a virtualenv
+          command: |
+            mkdir -p /tmp/venv/openfisca_france
+            python -m venv /tmp/venv/openfisca_france
+            echo "source /tmp/venv/openfisca_france/bin/activate" >> $BASH_ENV
+
+      - run:
+          name: Install dependencies
+          command: |
+            pip install --upgrade pip twine wheel
+            pip install --editable .[test] --upgrade
+          #  pip install --editable git+https://github.com/openfisca/openfisca-core.git@BRANCH_NAME#egg=OpenFisca-Core  # use a specific branch of OpenFisca-Core
+
+      - save_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+          paths:
+            - /tmp/venv/openfisca_france
+
+      - run:
+          name: Run tests
+          command: |
+            nosetests `circleci tests glob tests/**/*.py | circleci tests split`
+            openfisca-run-test `circleci tests glob tests/**/*.{yaml,yml} | circleci tests split`
+
+      - run:
+          name: Check version number has been properly updated
+          command: |
+            git fetch
+            .circleci/is-version-number-acceptable.sh
+
+  deploy_python3:
+    docker:
+      - image: python:3.6
+    environment:
+      PYPI_USERNAME: openfisca-bot  # Edit this value to replace it by your Pypi username
+      # PYPI_PASSWORD: this value is set in CircleCI's web interface; do not set it here, it is a secret!
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-py3-{{ checksum "setup.py" }}
+
+      - run:
+          name: Check for functional changes
+          command: if .circleci/detect-functional-changes.sh ; then circleci step halt ; fi
+
+      - run:
+          name: Upload a Python package to Pypi
+          command: |
+            source /tmp/venv/openfisca_france/bin/activate
+            .circleci/publish-python-package.sh
+
+workflows:
+  version: 2
+  build_and_deploy:
+    jobs:
+      - build_python2
+      - build_python3
+      - deploy_python2:
+          requires:
+            - build_python2
+            - build_python3
+          filters:
+            branches:
+              only: master
+      - deploy_python3:
+          requires:
+            - build_python2
+            - build_python3
+          filters:
+            branches:
+              only: master
```

Comment:
When I look at the workflow on circle ci, it seems that only 
```
 - build_python2
 - build_python3
```
are running and not 
```
- deploy_python2
 - deploy_python3
```
<img width="1677" alt="capture d ecran 2018-05-22 a 17 17 44" src="https://user-images.githubusercontent.com/13916213/40372178-7b334710-5de4-11e8-82d7-908be2412a85.png">

I don't thing it is due to the branch filtering because we had that in previous versions and the deployement phases did appear.

Reply:
Pretty sure it's due to branch filtering. See [here](https://circleci.com/workflow-run/689a7de6-5fee-4622-a876-0bc9b02f1146) and [here](https://circleci.com/workflow-run/bcea0d75-fba0-450a-bf45-370537f798f7) for Core examples.